internal/migrations/migrations.go

```diff
@@ -23,19 +23,36 @@ import (
 	"math/big"
 
 	"github.com/kwilteam/kwil-db/common"
+	"github.com/kwilteam/kwil-db/core/log"
 	"github.com/kwilteam/kwil-db/core/types/serialize"
 	"github.com/kwilteam/kwil-db/extensions/resolutions"
 )
 
+const (
+	StartMigrationEventType = "migration"
+)
+
+// migrator instance responsible for managing zero downtime migrations.
+var migrator *Migrator
```

Comment:
Seems unnecessary to make this a global singleton rather than an object passed around? AFAICT it is used in abci and the rpc service, both which take it as a dependency in their constructors. Furthermore, your `SetupMigrator` returns it.

Reply:
Hmm, I see that you have a `migration` resolution which starts the migration process. This seems related to another comment I made, where it seems like the more appropriate place to define a migration is with the genesis data?